src/Thumb.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <div :class="['thumb-wrapper', addClass, {'thumb-circle': isCircle}]" :style="[getBorder, getFontSize, getFontColor, getBgColor]">
+    <img ref="pic" @load.once="computeImgSize" :src="src" class="thumb-image" :width="computedWidth" :height="computedHeight" v-if="isImage">
+    <slot></slot>
+  </div>
+</template>
+
+<script>
+  const DEFAULT_SIZE = 100;
+
+  export default {
+    props: {
+      src: {
+        type: String
+      },
+      size: {
+        type: String,
+        default: ''
+      },
+      addClass: {
+        type: String,
+        default: ''
+      }, circle: {
+        type: Boolean,
+        default: false
+      },
+      border: {
+        type: String,
+        default: ''
+      },
+      text: {
+        type: String,
+        default: ''
+      },
+      fontsize: {
+        type: String,
+        default: ''
+      },
+      background: {
+        type: String,
+        default: ''
+      },
+      fontcolor: {
+        type: String,
+        default: ''
+      }
+    },
+    computed: {
+      isCircle () {
+        return this.circle !== undefined;
+      },
+      getBorder() {
+        // default is no border
+        return this.border === '' ? {} : {border: this.border};
+      },
+      getFontSize() {
+        // default is half of {size}
+        let fs = this.size/2;
+        fs = this.fontsize === '' ? fs : this.fontsize;
+        return {'font-size': fs + 'px'};
+      },
+      getFontColor() {
+        // default is no setting (inherit)
+        return this.fontcolor === '' ? {} : {'color': this.fontcolor};
+      },
+      getBgColor() {
+        // default is no setting (inherit)
+        return this.background === '' ? {} : {'background': this.background};
+      }
+    },
+    data () {
+      return {
+        computedWidth:'',
+        computedHeight:''
+      }
+    },
+    beforeCreate() {
+      console.log(this)
+    },
+    mounted() {
+      // TODO: Add text texts
```

Comment:
What do you mean by text texts? The caption?

Also, since this component is meant to contain images, we have to support `alt` text like `<pic>`. [Reference here](https://yoast.com/image-seo-alt-tag-and-title-tag-optimization/)

Reply:
ok, fixed in latest push